src/librustdoc/json/mod.rs

```diff
@@ -203,7 +202,7 @@ impl<'tcx> FormatRenderer<'tcx> for JsonRenderer<'tcx> {
         _diag: &rustc_errors::Handler,
     ) -> Result<(), Error> {
         debug!("Done with crate");
-        let mut index = (*self.index).clone().into_inner();
+        let mut index = self.index.clone();
```

Comment:
I'm not fully sure the behavior is the same here. It seems like before it was moving out of an `Rc`, which I didn't think was possible, so I suspect I'm misunderstanding what this is doing.

I wonder if we could avoid the clone altogether and just push directly into `self.index`, but I don't know this code, so that may break stuff.

Reply:
Okay, looks like I did change the behavior, because every test is panicking 😆

Reply:
It doesn't move outside the `Rc`, it just forces to dereference it and make it so that `clone` is called on the `RefCell` instead of the `Rc`. This code should have kept its meaning. 

What changed however is that now `clone`ing a `JsonRenderer` will create a totally separate `index`. In particular the `insert` in the `item` method will have effect only on `self`'s `index` while before it affected any `JsonRenderer` with the same origin.

Reply:
The only way for this to work is to remove all calls to `clone()`. If you get that to work I would expect it to have the same behavior as before.

Reply:
Thanks for the explanation! Probably this should wait then on #82356.